contracts/interchain-token-service/tests/flow_limit.rs

```diff
@@ -0,0 +1,290 @@
+mod utils;
+
+use axelar_gateway::types::Message as GatewayMessage;
+use axelar_soroban_std::{assert_contract_err, assert_invoke_auth_ok, events, traits::BytesExt};
+use interchain_token_service::{
+    error::ContractError,
+    event::FlowLimitSetEvent,
+    types::{HubMessage, InterchainTransfer, Message},
+    InterchainTokenServiceClient,
+};
+use soroban_sdk::testutils::{Address as _, Ledger as _};
+use soroban_sdk::{vec, xdr::ToXdr, Address, Bytes, BytesN, Env, String, Vec};
+use utils::{
+    approve_gateway_messages, register_chains, setup_env, setup_gas_token, setup_its_token,
+    HUB_CHAIN,
+};
+
+const TEST_FLOW_LIMIT: Option<i128> = Some(1000);
+const EPOCH_TIME: u64 = 6 * 60 * 60;
+
+fn setup_flow_limit(env: &Env, client: &InterchainTokenServiceClient) -> (BytesN<32>, Address) {
```

Comment:
call `setup_env` within this setup. This way only one setup call is needed per test